### PR TITLE
Migrate remaining ESIntegTestCases to internalClusterTest

### PR DESCRIPTION
**Pull request by rjernst**
This commit migrates the ESIntegTestCase tests in x-pack to the
internalClusterTest source set.

### CODE REVIEW

**Review by pugnascotia — APPROVED** [review]

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java [hunk 1]
```diff
@@ -80,7 +76,9 @@ public TestingConventionsTasks() {
         return getProject().getTasks()
             .withType(Test.class)
             .stream()
-            .filter(t -> tasks == null || tasks.contains(t.getName()))
+            // hardcoded to the old test task names, this will change as they are converted to individual source sets
```
Comment (mark-vieira) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java:79]: Ug, gross. What are we filtering _out_ here? Just internal cluster tests?

Comment (rjernst) [reply buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java:79]: Anything that is from a sourceSet *other* than test. The testingConventions task is really about the overlay of multiple test tasks on the test sourceset. In the previous PR I handled this by allowing to override the test tasks that should be inspected by it, but that turns out to be cumbersome when adding in x-pack projects. I'm open to other ideas...

Comment (mark-vieira) [reply buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java:79]: We could, when grabbing the candidate class files for a task, filter that collection for only those class files that belong to the test source set.

Comment (rjernst) [reply buildSrc/src/main/java/org/elasticsearch/gradle/precommit/TestingConventionsTasks.java]: Worked perfectly, thanks!

**Pull request merged**